### ISSUE TITLE
Installation stalled waiting for modules to be ready..

### ISSUE THREAD

**Issue by mikebaldry** [kalmhq/kalm#130]
It has done kalm-operator and cert-manager, and now it just sits there doing nothing. How can I help diagnose this issue?

**Comment by infrar** [comment]
hey @mikebaldry 

After installing kalm-operator and cert-manager, I believe the script will try to install istio. One way to diagnose the status of installing operators is to look at the status of namespaces with kubectl, for example:

`kubectl get ns`
`kubectl get all -n istio-system`

Example output from my dev machine:

![Screenshot from 2020-09-04 10-28-13](https://user-images.githubusercontent.com/67806321/92269774-b1494f00-ee99-11ea-8235-da44bca4f481.png)

**Comment by mikebaldry** [comment]
Yes, I see in the titlebar it's waiting for the istio stuff to appear. After inspecting my cluster, I don't have an `istio-system` at all. It has been running for hours now, so I guess I'll stop it and run again, see what happens? 

When I look at the logs of kalm operator, I see this error repeated..

`controllers.KalmOperatorConfig	install certManager error.	{"kalmoperatorconfig": "kalm-operator/kalmoperatorconfig-sample", "error": "CustomResourceDefinition.apiextensions.k8s.io \"certificaterequests.cert-manager.io\" is invalid: spec.conversion.webhookClientConfig: Required value: required when strategy is set to Webhook, but not allowed because the CustomResourceWebhookConversion feature is disabled"}
[kalm-operator-6fd85d694-hl6gw manager] github.com/go-logr/zapr.(*zapLogger).Error`

**Comment by davidqhr** [comment]
It seems your k8s cluster version is low, some feature is not enabled by default. Can you try kalm on a k8s cluster which version is not less than 1.15?

**Comment by mikebaldry** [comment]
Ah yes I'm on 1.14. it suggests in the docs it's supported. What do I need to enable? I'm happy to upgrade my cluster for fun and it's only running a few things which I can easily port. I feel it'd be useful to define what's required to get it working in 1.14 if it is supported though

Many thanks

**Comment by Linutux** [comment]
I have had the same issue, the install being stuck. I had nodes with 2 vCPUs and 2GB of RAM for this. I upgraded them to 4 vCPUs and 8GB of RAM. Then the installation went smoothly. Afterwards I downgraded the nodes again.

**Comment by davidqhr** [comment]
Thanks for the feedback. Sorry, it seems that the document is not clear enough. The installation failure of 1.14 should be mainly because some of the features that kalm depends on are not enabled on your cluster. In your case, it should be CustomResourceWebhookConversion.

1. I will improve the documentation later, suggesting how to configure the apiserver parameters on the lower version of k8s to ensure that the features required by the kalm function are turned on.
1. Remind everyone to try kalm with a suitable machine configuration, for example (2Core, 4G Memory)
